readthedocs/notifications/storages.py

```diff
@@ -0,0 +1,45 @@
+from django.contrib.messages.storage.base import Message
+from django.utils.safestring import mark_safe
+from messages_extends.storages import FallbackStorage
+from messages_extends.models import Message as PersistentMessage
+from messages_extends.constants import PERSISTENT_MESSAGE_LEVELS
+
+
+class FallbackUniqueStorage(FallbackStorage):
+
+    """Persistent message fallback storage, but only stores unique notifications
+
+    This loops through all backends to find messages to store, but will skip
+    this step if the message already exists for the user in the database.
+
+    This class also assumes that notifications set as persistent messages are
+    more likely to have HTML that should be marked safe. If the level matches a
+    persistent message level, mark the message text as safe so that we can
+    render the text in templates.
+    """
```

Comment:
Would be good to explain why this is needed. Is it to avoid showing the user the same message twice/a million times? :)